build.gradle

```diff
@@ -419,6 +419,22 @@ subprojects {
         "-Xlint:constant",
         "-Xlint:unused"
       ]
+
+      // Inline more aggressively when compiling the `core` jar since it's not meant to be used as a library.
+      // More specifically, inline classes from the Scala library so that we can inline methods like `Option.exists`
+      // and avoid lambda allocations. This is only safe if the Scala library version is the same at compile time
+      // and runtime. We cannot guarantee this for libraries like kafka streams, so only inline classes from the
+      // Kafka project in that case.
+      List<String> inlineFrom
+      if (project.name.equals('core'))
+        inlineFrom = ["-opt-inline-from:scala.**", "-opt-inline-from:kafka.**", "-opt-inline-from:org.apache.kafka.**"]
```

Comment:
@ijuma I apologize if this the wrong place to ask a question about this PR. What would be the best place to start a discussion on removing `"-opt-inline-from:scala.**"` or reducing the scope of the glob to only include relevant methods like `Option.exists`? I'm not so familiar with the Kafka development process.

At Twitter, we are currently unable to upgrade to 2.5.x because kafka_2.12:2.5.1 inlines anonymous classes from the 2.12.11 library that no longer exist in the 2.12.12 library, resulting in crashes like this
```
 java.lang.NoClassDefFoundError: scala/math/Ordering$$anon$7
```
We can work around this issue by publishing a custom build of Kafka but that's not a sustainable solution.

These inlining settings are causing issues for a lot of users based on my search online. Even members of the Scala compiler team who worked on the inliner have suggested that the inliner settings in the Kafka build are based on false assumptions https://github.com/akka/alpakka-kafka/pull/1212#issuecomment-706933445 cc/ @lrytz

Reply:
You can submit a PR. :)

Reply:
I opened https://github.com/apache/kafka/pull/9548 :)

Reply:
Let's continue the discussion in #9548. Note the comment from @lrytz is regarding the claim that the `core` module should not be used as a library. This is the recommendation from the Kafka project though, so we can discuss in #9548 why users are depending on this module as a library.